java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -182,15 +102,30 @@ public ColumnVector(DType type, long rows, Optional<Long> nullCount,
     incRefCountInternal(true);
   }
 
+  /**
+   * Create a new column vector based off of data already on the device with child columns.
+   * @param type the type of the vector, typically a nested type
+   * @param rows the number of rows in this vector.
+   * @param nullCount the number of nulls in the dataset.
+   * @param dataBuffer the data stored on the device.  The column vector takes ownership of the
+   *                   buffer.  Do not use the buffer after calling this.
+   * @param validityBuffer an optional validity buffer. Must be provided if nullCount != 0. The
+   *                      column vector takes ownership of the buffer. Do not use the buffer
+   *                      after calling this.
+   * @param offsetBuffer a host buffer required for strings and string categories. The column
+   *                    vector takes ownership of the buffer. Do not use the buffer after calling
+   *                    this.
+   * @param nestedColumnVectors the child columns list
+   */
   public ColumnVector(DType type, long rows, Optional<Long> nullCount,
-                      DeviceMemoryBuffer dataBuffer, DeviceMemoryBuffer validityBuffer,
```

Comment:
Why did this change?  `DeviceMemoryBuffer` implies that it owns the data so the ColumnVector can take ownership of the data and not have to worry about it.  The other `BaseDeviceMemoryBuffer` child is `DeviceMemoryBufferView` which does not own the data at all so there is no guarantee about the life cycle of the data involved.  I am OK with this in he right situations, but we cannot say that the ColumnVector takes ownership of the data, because there are cases where the buffer does not even own the data so it cannot take ownership.

Reply:
I was forced to make this change due to how offheap returns the base version. To copy the view into a vector, maybe then i should allocate new device buffers and pass that in? That makes more sense now that I think about it.

Reply:
Unless we want to change everything that expects to operate on `ColumnVector` to instead expect a `ColumnView` we would need two types of ways to create a ColumnVector from a column view:
1. ColumnVector copies the data from a view (for example, extracting a child from a nested type column so it can outlive the parent column)
1. ColumnVector wraps the view and tracks it (much like it does today when building columns from a contiguous table)

Ultimately I think it would be cleaner to have ColumnVector only ever copy the data when being constructed by a view, but then that means everyone needs to start wielding ColumnView when they have inputs rather than ColumnVector.

Reply:
@jlowe I think I address 1. not sure if the current change will do what u expect for contig split in 2. Let me know what you think and if we want to add one more constructor to ColumnView.

Reply:
@revans2 
> Why did this change? 
>
I have undone this change.

Reply:
> @jlowe I think I address 1.

I didn't see the constructor or factory method (probably the latter is more appropriate) that creates a new `ColumnVector` from a view.  Apologies if I missed it.

> not sure if the current change will do what u expect for contig split in 2.

For contiguous split, we're probably going to leverage view handles rather than full view objects, so as long as there's still `ColumnVector.fromViewWithContiguousAllocation` it should be OK.

Reply:
@jlowe , I agree I think I should rename this and make it a factory method ti make it clearer and visible -
```
public final ColumnVector copyToColumnVector() {
    long mainCvHandle = ColumnView.copyColumnViewToCV(viewHandle);
    return new ColumnVector(mainCvHandle);
  }
```

Reply:
Also javadoc! Will add.

Reply:
Done, @jlowe . I have done the following as a static factory method in ColumnVector instead of a class method in ColumnView:

```
  /** Creates a ColumnVector from a column view handle
   * @return a new ColumnVector
   */
  public static ColumnVector fromColumnView(long columnViewHandle) {
    long mainCvHandle = ColumnView.copyColumnViewToCV(columnViewHandle);
    return new ColumnVector(mainCvHandle);
  }
```